vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -520,7 +521,8 @@ void propagatePencil(Vec* dz, Vec* values, const uint dimension,
    const uint nTargetNeighborsPerPencil = 1;
 
    // Vector buffer where we write data, initialized to 0*/
-   Vec targetValues[(lengthOfPencil + 2 * nTargetNeighborsPerPencil) * WID3 / VECL];
+   std::vector<Vec, aligned_allocator<Vec,64>> targetValues;
+   targetValues.reserve((lengthOfPencil + 2 * nTargetNeighborsPerPencil) * WID3 / VECL);
```

Comment:
These two lines can also be combined by simply giving the size as the constructor argument, right?
I'm kinda wondering if the compiler manages to optimize this away, or whether this will lead to one extraneous allocation here.

Reply:
Sure, if you think that is safer. I was not sure how it treats the custom allocator when reallocating the vector.